python_modules/dagster/dagster/_core/execution/context/hook.py

```diff
@@ -205,11 +205,11 @@ def temp_graph():
 
             self._op = temp_graph.solids[0]
 
-        # Open resource context manager
-        self._resource_defs = wrap_resources_for_execution(resources)
-        self._resources_cm = build_resources(self._resource_defs)
-        self._resources = self._resources_cm.__enter__()  # pylint: disable=no-member
-        self._resources_contain_cm = isinstance(self._resources, IContainsGenerator)
+        UsesResourcesContext.__init__(
+            self,
+            resources=resources,
+            context_str="hook",
+        )
```

Comment:
what is going on here? why cant we just use `super()` ?

Reply:
We can't use super because the two constructors have different arguments. It's a case of diamond inheritance: https://stackoverflow.com/questions/46199975/resolving-diamond-inheritance-within-python-classes

Reply:
i dont think its a diamond since `UsesResourcesContext` doesn't have a parent class - look in to the module resolution order and maybe switch up the order of inheritance - the `mro()` method can be useful for debugging. If that proves fruitless these deserve a comment at least 

https://www.python.org/download/releases/2.3/mro/